src/components/WebCLI/connection.ts

```diff
@@ -3,84 +3,119 @@ type Options = {
   messageCallback: (message: string) => void;
   onclose: () => void;
   onopen: () => void;
+  onerror: (error: Event | string) => void;
 };
 
 class Connection {
-  address: string;
-  _messageBuffer = "";
-  _timeout: number | null = null;
-  _messageCallback: (message: string) => void;
-  _wsOnOpen: () => void;
-  _wsOnClose: () => void;
-  _ws: WebSocket | null = null;
+  #address: string;
+  #messageBuffer = "";
+  #timeout: number | null = null;
+  #messageCallback: (message: string) => void;
+  #wsOnOpen: () => void;
+  #wsOnClose: () => void;
+  #wsOnError: (error: Event | string) => void;
+  #ws: WebSocket | null = null;
 
   constructor(options: Options) {
-    this._messageCallback = options.messageCallback;
-    this.address = options.address;
-    this._wsOnOpen = options.onopen;
-    this._wsOnClose = options.onclose;
+    this.#messageCallback = options.messageCallback;
+    this.#address = options.address;
+    this.#wsOnOpen = options.onopen;
+    this.#wsOnClose = options.onclose;
+    this.#wsOnError = options.onerror;
+  }
+
+  getAddress() {
```

Comment:
Not sure how useful this getter is, considering that we are returning the address itself without any modifications. Should we make `#address` public instead and directly access it in `components/WebCLI/WebCLI.tsx`?

Reply:
Thanks for the suggestion!